lib/global_path.rb

```diff
@@ -4,7 +4,7 @@ def path(p)
   end
 
   def cdn_path(p)
-    "#{GlobalSetting.cdn_url}#{path(p)}"
+    GlobalSetting.cdn_url&.empty? ? p : "#{GlobalSetting.cdn_url}#{path(p)}"
```

Comment:
Thanks to Rails's magic, this can be reduced to 

```suggestion
"#{GlobalSetting.cdn_url.presence}#{path(p)}"
```